fyrox-ui/src/widget.rs

```diff
@@ -550,6 +559,166 @@ impl WidgetMessage {
+            .with_background_hover(brush)
+    }
+
+    /// Sets the normal background color to the provided brush
+    pub fn with_background_normal(mut self, brush: Brush) -> Self {
+        self.background_normal = Some(brush);
+        self
+    }
+
+    /// Sets the hover background color to the provided brush
+    pub fn with_background_hover(mut self, brush: Brush) -> Self {
+        self.background_hover = Some(brush);
+        self
+    }
+
+    /// Sets the normal and hover foreground colors to the provided brush
+    #[inline]
+    pub fn with_foreground(self, brush: Brush) -> Self {
+        self.with_foreground_normal(brush.clone())
+            .with_foreground_hover(brush)
+    }
+
+    /// Sets the normal foreground color to the provided brush
+    pub fn with_foreground_normal(mut self, brush: Brush) -> Self {
+        self.foreground_normal = Some(brush);
+        self
+    }
+
+    /// Sets the hover foreground color to the provided brush
+    pub fn with_foreground_hover(mut self, brush: Brush) -> Self {
+        self.foreground_hover = Some(brush);
+        self
+    }
+}
+
+impl WidgetPaletteBuilder {
+    /// Builds a `WidgetPalette` from this builder
+    pub fn build(self) -> WidgetPalette {
+        let background_normal = self
+            .background_normal
+            .unwrap_or_else(|| BRUSH_PRIMARY.clone());
+
+        let background_hover = self
+            .background_hover
+            .unwrap_or_else(|| background_normal.clone());
+
+        let foreground_normal = self
+            .foreground_normal
+            .unwrap_or_else(|| BRUSH_FOREGROUND.clone());
+
+        let foreground_hover = self
+            .foreground_hover
+            .unwrap_or_else(|| foreground_normal.clone());
+
+        WidgetPalette {
+            background_normal,
+            background_hover,
+            foreground_normal,
+            foreground_hover,
+        }
+    }
+}
+
+impl Default for WidgetPalette {
+    fn default() -> Self {
+        Self {
+            background_normal: BRUSH_PRIMARY.clone(),
+            background_hover: BRUSH_PRIMARY.clone(),
+            foreground_normal: BRUSH_FOREGROUND.clone(),
+            foreground_hover: BRUSH_FOREGROUND.clone(),
+        }
+    }
+}
+
+/// A set of messages that is used to modify [`WidgetPalette`] states.
+#[derive(Debug, Clone, PartialEq)]
+pub enum WidgetPaletteMessage {
+    /// Set the palette normal background brush.
+    BackgroundNormal(Brush),
+    /// Set the palette hover background brush.
+    BackgroundHover(Brush),
+    /// Set the palette normal foreground brush.
+    ForegroundNormal(Brush),
+    /// Set the palette hover foreground brush.
+    ForegroundHover(Brush),
+}
+
+impl WidgetPalette {
+    /// Updates the background and foreground color of the provided
+    /// node to be the normal colors from this palette
+    pub fn set_normal(&self, handle: Handle<UiNode>, ui: &mut UserInterface) {
+        ui.send_message(WidgetMessage::background(
```

Comment:
From looking at the code I've just realized there isn't actually any need for me to be sending messages here as it can just have access to the widget itself instead and update it from there, the new code:
![image](https://github.com/FyroxEngine/Fyrox/assets/33708767/a7bf3127-188c-4cb6-bd98-77d28535bd7a)